bokehjs/src/coffee/api/typings/models/sources.d.ts

```diff
@@ -5,10 +5,13 @@ declare namespace Bokeh {
         callback: Callback;
     }
 
+    type Data = {[key: string]: ArrayLike<any>};
+
     export var ColumnDataSource: { new(attributes?: KeyVal, options?: KeyVal): ColumnDataSource };
     export interface ColumnDataSource extends DataSource {
-        data: {[key: string]: ArrayLike<any>};
-        stream(new_data: Object, rollover: Number): void;
+        data: Data;
+
+        stream(new_data: Data, rollover: number): void;
```

Comment:
As to `number` vs `Number` I'm not sure if it really matters, but `number` seems more natural.